### PR TITLE
JPN-375 Tracking for community page

### PR DESCRIPTION
**Pull request by perjg**

### CODE REVIEW

File: extensions/wikia/CommunityPage/scripts/ext.communityPage.js [hunk 1]
```diff
@@ -170,18 +178,52 @@ require([
...
+					label: data,
+				});
+			}
+		});
+	}
+
+	function initModalTracking() {
+		// Track clicks in contribution modal
+		$('#CommunityPageModalDialog').on('mousedown touchstart', 'a', function (event) {
+			var data = $(event.currentTarget).data('tracking');
```
Comment (Grunny) [inline extensions/wikia/CommunityPage/scripts/ext.communityPage.js]: This handler function is exactly the same as the one for the module tracking above, so I'd move it out into a `handleClick` method and then reuse it for both.

File: extensions/wikia/CommunityPage/templates/topContributorsModal.mustache [hunk 2]
```diff
@@ -16,10 +16,10 @@
 	{{#contributors}}
 		<li>
 			<div class="avatar-container">
-				<a href="{{profilePage}}">{{{avatar}}}</a>
+				<a data-tracking="top-contributors-modal-avatar" href="{{profilePage}}">{{{avatar}}}</a>
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributorsModal.mustache]: I think top-contributors-modal-user-avatar will be more clear

File: extensions/wikia/CommunityPage/templates/topContributorsModal.mustache [hunk 3]
```diff
@@ -16,10 +16,10 @@
...
 			</div>
 			<div class="contributor-details">
-				{{count}}. <a href="{{profilePage}}">{{userName}}</a>
+				{{count}}. <a data-tracking="top-contributors-modal" href="{{profilePage}}">{{userName}}</a>
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributorsModal.mustache]: I think top-contributors-modal-user will be more clear

File: extensions/wikia/CommunityPage/templates/topContributors.mustache [hunk 4]
```diff
@@ -19,7 +19,7 @@
 				<a href="{{profilePage}}">{{{avatar}}}</a>
 			</div>
 			<div class="contributor-details">
-				{{count}}. <a href="{{profilePage}}">{{userName}}</a>
+				{{count}}. <a data-tracking="top-contributors" href="{{profilePage}}">{{userName}}</a>
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: I think format like top-contributors-{modal/module}-user will be more clear which means we would need to provide different label depending whether topContributors template is used in module or modal.

What do you think about it?

File: extensions/wikia/CommunityPage/templates/topContributors.mustache [hunk 5]
```diff
@@ -19,7 +19,7 @@
 				<a href="{{profilePage}}">{{{avatar}}}</a>
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: Please add tracking for avatar as well

File: extensions/wikia/CommunityPage/templates/recentlyJoined.mustache [hunk 6]
```diff
@@ -2,10 +2,10 @@
...
 	{{#members}}
 		<div class="avatar-container">
-			<a href="{{profilePage}}">{{{avatar}}}</a>
+			<a data-tracking="recently-joined-avatar" href="{{profilePage}}">{{{avatar}}}</a>
 		</div>
 	{{/members}}
 {{/haveMembers}}
 <span class="recently-joined">
-	<a href="#" id="viewAllMembers">{{allMembers}}</a>
+	<a data-tracking="show-modal" href="#" id="viewAllMembers">{{allMembers}}</a>
```
Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/recentlyJoined.mustache]: In upcoming https://github.com/Wikia/app/pull/10399 we'll adding new entry point to open modal. What about changing the label to tell which entry point is clicked? e.g. show-modal-all

**Comment by kamilkoterba** [comment]
@grunny, @adamk, @lukaszk, @perjg Second thought on labels. Initially we decided with Per to add one category for community-page and than distinguish by labels. Although I think it may get complicated with time. What do you think about add categories per UI group:
- community-page-contributors-module
- community-page-contributors-modal
- community-page-header
- community-page-recent-activity
- etc.

File: extensions/wikia/CommunityPage/templates/topContributors.mustache (hunk 4 above)
Comment (perjg) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: There are already two templates for this: topContributors.mustache and topContributorsModal.mustache. The latter is used in the modal only and has the tracking label top-contributors-modal.

Comment (perjg) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: @kamilkoterba Right now, the special tracking has the tag -modal added, with no suffix for the main module. I'm ok with adding -module to the ones in the main module. I'm not so sure about adding -user to the end of everything.

Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: Cool.
I would add user only to links that redirects to user profile page. What are your concerns?

Comment (perjg) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: Ok, that makes sense. I misunderstood your intention.

Comment (kamilkoterba) [inline extensions/wikia/CommunityPage/templates/topContributors.mustache]: 👍

**Comment by kamilkoterba** [comment]
LGTM. moving o testing

**Comment by kamilkoterba** [comment]
Testing passed.

**Pull request merged**